Add German translations (DE)

SuriyaaKudoIsc opened the pull request:
@sferik:

Hi guys, :smile:

I *add the first translations* in **German** in **`config/locales/de.yml`**. :pencil2:

Don't worry. More code is coming! ;-) I'm [currently translating](https://github.com/SuriyaaKudoIsc/rubygems.org/tree/patch-2) the rest. :-)


*Yours,*
[**Suriyaa Kudo**](https://github.com/SuriyaaKudoIsc) :octocat:

farukaydin commented:
@SuriyaaKudoIsc can you squash commits? After you complete translations, maybe @sferik or other german speakers can review it.

Thanks for PR

SuriyaaKudoIsc commented:
@farukaydin: I'm not finished. Please [review this](https://github.com/SuriyaaKudoIsc/rubygems.org/blob/patch-2/config/locales/de.yml) after I'm finished. I'll notify you all.

sferik commented:
@SuriyaaKudoIsc For pull requests that are still a work in progress, it’s common to add [WIP] to the title, so other know to wait to review it. I have gone ahead and made this change for you. When you are finished, please remove the [WIP] tag.

SuriyaaKudoIsc commented:
:+1: Ok.

SuriyaaKudoIsc commented:
I'm finished. Review the final code at https://github.com/SuriyaaKudoIsc/rubygems.org/blob/patch-2/config/locales/de.yml.

dwradcliffe commented:
Please squash your commits into one commit.

arthurnn commented:
thanks @SuriyaaKudoIsc for you work here. I will wait for @sferik to merge this. also if @fw42 fells like reviewing some i18n.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+  title: "RubyGems.org"
+  subtitle: "Deine Community des Gem-Hostingservices"
+  help_url: "http://help.rubygems.org"
+  please_sign_up: "Zugriff verweigert. Bitte melden Sie in https://rubygems.org mit einem Konto an"
```
This sounds weird. First of all, the `subtitle` above uses "Deine" (colloquial) but here you use "Sie" (polite). Should be consistent. But the grammer is wrong too. Something like "Bitte melden Sie sich unter ... an" sounds more natural to me.

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+  sign_in: "Anmelden"
+  sign_up: "Registrieren"
+  locale_name: "German"
+  this_rubygem_could_not_be_found: "Dieser Ruby Gem konnte nicht gefunden werden."
```
I guess we have to discuss if it's "der Gem" or "das Gem" (you are using "der", but personally I find that "das" sounds more correct).

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+    show:
+      latest: "Letzte Aktualisierungen"
+      latest_title: "Neueste Aktualisierungen RSS Feed"
+      no_subscriptions: "Du hast noch keine Gems abonniert. Besuche %{gem_link} um ein Gem zu abonnieren!"
```
here ("ein Gem") you are using "das Gem" (otherwise it would be "einen Gem"). Should be consistent (see above).

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+      no_subscriptions: "Du hast noch keine Gems abonniert. Besuche %{gem_link} um ein Gem zu abonnieren!"
+      gem_link_text: "Gem-Seite"
+      mine: "Meine Gems"
+      no_owned: "Du hast noch keine Gems gepusht. Schau doch vielleicht die Dokumentation auf %{creating_link} an und %{migrating_link} ein Gem from RubyForge."
```
This sounds like the second part of the second sentence is missing a verb?

SuriyaaKudoIsc replied:
In which line?

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+        sign_in: "Anmelden"
+        sign_up: "Registrieren"
+        sign_out: "Abmelden"
+        search_gem: "Suche Gems&hellip;"
```
ugh HTML markup in YML files is kinda ugly, is this necessary?

SuriyaaKudoIsc replied:
I use the **`&hellip;`** code from the **`en.yml`** file.

sferik replied:
I just noticed you need to remove the space between the `&` and `hellip;`.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+        optimized_by: "Optimiert mit"
+        tracking_by: "Getrackt von"
+        monitored_by: "Überwacht von"
+        gems_served_by: "Gems serviert von"
```
"serviert" is for food... I would say "angeboten von" maybe?

SuriyaaKudoIsc replied:
:+1: Sounds better.

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+    about:
+      title: Über uns
+      purpose:
+        header: "Willkommen in %{site}, die Ruby-Community für Gem-Hostingservice. Die Gründe sind vielfältig:"
```
`s/in/auf/`. And "Ruby-Community für Gem-Hostingservice" makes no sense. What are you trying to say?

SuriyaaKudoIsc replied:
Actually I mean: "**Community für Ruby Gem-Hostingservice**".

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+      title: Über uns
+      purpose:
+        header: "Willkommen in %{site}, die Ruby-Community für Gem-Hostingservice. Die Gründe sind vielfältig:"
+        better_api: Bereitstellen eines besseren APIs für den Umgang mit Gems
```
`s/eines/einer/`, `s/APIs/API/`

SuriyaaKudoIsc replied:
What would you choose?

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+        transparent_pages: Erstellen transparenter und zugänglicher Projektseiten
+        enable_community: Der Gemeinschaft ermöglichen, die Webseite zu verbessern und zu erweitern
+      founding: "Das Projekt wurde im April 2009 von %{founder} gestartet und hat seitdem über %{contributors} und %{downloads}. Ab dem Release von RubyGems 1.3.6 wurde die Website von Gemcutter zu %{title} umbenannt, um eine zentrale Rolle in der Ruby-Community zu verfestigen."
+      support: "Obwohl RubyGems.org nicht von einem bestimmten Unternehmen ausgeführt wird, haben schon viele uns bisher geholfen. Das aktuelle Design, die Illustrationen und die Front-End-Entwicklung für dieser Webseite wurden von %{dockyard} erstellt. %{github} hat auch von unschätzbarem Wert uns geholfen den Code mit anderen Entwicklern zuteilen, um an dem Code leichter zusammenarbeiten. Die W ...
```
I would change "haben schon viele uns bisher geholfen" to "haben viele Unternehmen uns bereits unterstützt"

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+        enable_community: Der Gemeinschaft ermöglichen, die Webseite zu verbessern und zu erweitern
+      founding: "Das Projekt wurde im April 2009 von %{founder} gestartet und hat seitdem über %{contributors} und %{downloads}. Ab dem Release von RubyGems 1.3.6 wurde die Website von Gemcutter zu %{title} umbenannt, um eine zentrale Rolle in der Ruby-Community zu verfestigen."
+      support: "Obwohl RubyGems.org nicht von einem bestimmten Unternehmen ausgeführt wird, haben schon viele uns bisher geholfen. Das aktuelle Design, die Illustrationen und die Front-End-Entwicklung für dieser Webseite wurden von %{dockyard} erstellt. %{github} hat auch von unschätzbarem Wert uns geholfen den Code mit anderen Entwicklern zuteilen, um an dem Code leichter zusammenarbeiten. Die W ...
+      technical: "Einige Einblicke in die technischen Aspekte dieser Webseite: Es ist 100% in Ruby geschrieben. Der Hauptseite ist verwendet die %{rails}-Software. RubyGem nutzt %{sinatra}, um die Gems bereitzustellen. Die Gems werden in %{s3} gehostet und die Zeit zwischen der Veröffentlichung eines neuen Gems und nachdem er bereit zur Installation freigegeben ist, ist minimal. Für mehr Informat ...
```
`s/Der Hauptseite/Die Hauptseite/`

SuriyaaKudoIsc replied:
:+1: Done. Change it to "**Die Hauptseite**".

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+  profiles:
+    edit:
+      title: Bearbeite Profil
+      hide_email: Verberge E-Mail in öffentliches Profil
```
`s/öffentliches/öffentlichem/`

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+      api_access:
+        title: API-Zugang
+        key_is: "Dein API-Schlüssel ist %{key}."
+        credentials: "Wenn du %{gem_commands_link} über das Terminal oder über das CMD verwenden möchtest, dann brauchst du die %{gem_credentials_file}-Datei, die du mit dem folgenden Befehl erzeugen kannst:"
```
"das CMD"? What is that?

SuriyaaKudoIsc replied:
The "**command prompt**".

SuriyaaKudoIsc replied:
:+1: Done. Change "*das CMD*" to "**die Kommandozeile**".

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+      title: 'Gems'
+      downloads: Downloads
+    show:
+      not_hosted_notice: Dieses Gem wird nicht aktuell auf RubyGems.org gehostet.
```
`s/nicht aktuell/aktuell nicht/`

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+    show:
+      not_hosted_notice: Dieses Gem wird nicht aktuell auf RubyGems.org gehostet.
+      install: installieren
+#      yanked_notice: "This gem has been yanked, and it is not available for download directly or for other gems that may have depended on it."
```
?

SuriyaaKudoIsc replied:
Correct confirm_reset

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+#      yanked_notice: "This gem has been yanked, and it is not available for download directly or for other gems that may have depended on it."
+      authors_header: Autoren
+      downloads_for_this_version: "Für diese Version"
+      owners_header: Besitzern
```
Depends on the context of where this is inserted, but this should probably say "Besitzer" (no trailing "n")

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+    index:
+      total_downloads: Downloads insgesamt
+      total_gems: Gems insgesamt
+      total_users: Gems insgesamt
```
"Benutzer insgesamt"?

SuriyaaKudoIsc replied:
Ops. :+1: Correct it.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+
+  users:
+    new:
+      have_account: "Hast du schon ein Account?"
```
`s/ein/einen/`

SuriyaaKudoIsc replied:
:+1: Done.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+      versions_since: "%{count} Versionen seit %{since}"
+      not_hosted_notice: Dieses Gem wird nicht gerade von RubyGems.org gehostet.
+    version:
+#      yanked: "yanked"
```
?

SuriyaaKudoIsc replied:
Where is the mistake?

sferik replied:
Yanked appears to be commented out.

SuriyaaKudoIsc replied:
@sferik @fw42: What's the perfect German translation for "**`yanked`**"?

arthurnn replied:
if yanked is the same word in germany or we are not sure, i would uncoment this line and we are good to go!

SuriyaaKudoIsc replied:
@sferik @arthurnn: :+1: I commented the word "yanked" out and squashed the commit, too:

```bash
Windows PowerShell
Copyright (C) 2015 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\Suriyaa\Documents\GitHub> git clone https://github.com/SuriyaaKudoIsc/rubygems.org.git
Cloning into 'rubygems.org'...
remote: Counting objects: 31017, done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 31017 (delta 4), reused 0 (delta 0), pack-reused 31007
Receiving objects: 100% (31017/31017), 8.26 MiB | 3.22 MiB/s, done.
Resolving deltas: 100% (20330/20330), done.
Checking connectivity... done.
C:\Users\Suriyaa\Documents\GitHub> cd rubygems.org
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [master]> git checkout patch-2
Branch patch-2 set up to track remote branch patch-2 from origin.
Switched to a new branch 'patch-2'
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [patch-2]> git rebase -i master
[detached HEAD 55f6190] Create de.yml
 Author: Suriyaa Kudo <SuriyaaKudoIsc@users.noreply.github.com>
 Date: Sun Dec 6 13:08:35 2015 +0100
 1 file changed, 203 insertions(+)
 create mode 100644 config/locales/de.yml
Successfully rebased and updated refs/heads/patch-2.
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [patch-2]> git push -f origin patch-2
Counting objects: 5, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 3.65 KiB | 0 bytes/s, done.
Total 5 (delta 3), reused 0 (delta 0)
To https://github.com/SuriyaaKudoIsc/rubygems.org.git
 + e8501d5...55f6190 patch-2 -> patch-2 (forced update)
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [patch-2]>
```

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+      user:
+        avatar: Avatar
+        email: E-Mail-Adresse
+        handle: Verhandele
```
yeah no :) this is the german word for "to handle", but not for the noun "handle". This should be something like "Kürzel" or "Benutzername" (username).

SuriyaaKudoIsc replied:
:+1: Ok. Correct it.

fw42 commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+        mail: Mailingliste URL
+        wiki: Wiki URL
+      session:
+        who: E-Mail oder Verhandele
```
same re: handle

SuriyaaKudoIsc replied:
:+1: Done.

arthurnn commented:
❤️  @fw42

SuriyaaKudoIsc commented:
I correct everything. :pencil2:

> See https://github.com/SuriyaaKudoIsc/rubygems.org/blob/patch-2/config/locales/de.yml. 

Thanks for your help @fw42. :smile:

sferik commented:
Das sieht gut aus für mich. Ich möchte nur bitten Sie, die vielen Commits in einem einzigen Squash begehen.

SuriyaaKudoIsc commented:
@sferik Mit welchen Git-Command kann ich das machen?

sferik commented:
Auf Ihrer Branche `patch-2`:

```
git rebase -i master
```

Du wirst sehen:
```
pick 424f7df Create de.yml
pick 8216a66 Add title
pick 4516dd7 Add subtitle
pick 8f41c70 Add help_url
⋮
pick f4dffea Correct who
```

Ändern Sie jeden "pick" auf "fixup", mit Ausnahme der ersten begehen.
```
pick  424f7df Create de.yml
fixup 8216a66 Add title
fixup 4516dd7 Add subtitle
fixup 8f41c70 Add help_url
⋮
fixup f4dffea Correct who
```
Dann speichern und zu beenden.

SuriyaaKudoIsc commented:
@sferik: You mean **`git push -f origin patch-2`** not *`git push -f origin branch-2`*. XD

SuriyaaKudoIsc commented:
I run:

```bash
Windows PowerShell
Copyright (C) 2015 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\Suriyaa\Documents\GitHub> git clone https://github.com/SuriyaaKudoIsc/rubygems.org.git
Cloning into 'rubygems.org'...
remote: Counting objects: 31357, done.
remote: Compressing objects: 100% (80/80), done.
remote: Total 31357 (delta 60), reused 0 (delta 0), pack-reused 31277
Receiving objects: 100% (31357/31357), 8.29 MiB | 2.68 MiB/s, done.
Resolving deltas: 100% (20598/20598), done.
Checking connectivity... done.
C:\Users\Suriyaa\Documents\GitHub> cd rubygems.org
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [master]> git checkout patch-2
Branch patch-2 set up to track remote branch patch-2 from origin.
Switched to a new branch 'patch-2'
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [patch-2]> git rebase -i master
[detached HEAD 37bfa6a] Create de.yml
 Author: Suriyaa Kudo <SuriyaaKudoIsc@users.noreply.github.com>
 Date: Sun Dec 6 13:08:35 2015 +0100
 1 file changed, 204 insertions(+)
 create mode 100644 config/locales/de.yml
Successfully rebased and updated refs/heads/patch-2.
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [patch-2]> git push -f origin patch-2
Counting objects: 5, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 3.72 KiB | 0 bytes/s, done.
Total 5 (delta 3), reused 1 (delta 0)
To https://github.com/SuriyaaKudoIsc/rubygems.org.git
 + f4dffea...37bfa6a patch-2 -> patch-2 (forced update)
C:\Users\Suriyaa\Documents\GitHub\rubygems.org [patch-2]>
```

Edit the file:

```editor
pick ad61f0a Add a company name for S3
pick 424f7df Create de.yml
fixup 8216a66 Add title
fixup 4516dd7 Add subtitle
fixup 8f41c70 Add help_url
fixup ce552f6 Add please_sign_up
fixup af77183 Add feed_subscribed
fixup 0807f86 Add feed_latest
fixup c58556b Add time_ago
fixup c1f649b Add form_disable_with
fixup bbdfc5b Add update
fixup f7c0973 Add edit
fixup a5d7b05 Add sign_in
fixup 6af4c03 Add sign_up
fixup cbd9c72 Add locale_name
fixup e2f6e88 Add this_rubygem_could_not_be_found
fixup 11c2c91 Add dashboard, show & latest
fixup c4cbb7d Add latest_title
fixup c0dcd75 Add no_subscriptions
fixup 216529e Add gem_link_text
fixup 8a406f5 Add mine
fixup 9a9a1d1 Add no_owned
fixup 86b092b Add creating_link_text
fixup 053cc0f Add migrating_link_text
fixup 036021c Add my_subscriptions
fixup 391509e Add home, index & downloads_counting
fixup cd97f12 Add find_blurb
fixup 64e3ffa Add learn & install_rubygems
fixup 1fdc091 Add layouts, application, header & dashboard
fixup 3d67300 Add sign_in
fixup 6d28ae5 Add sign_up
fixup f7b6a4b Add sign_out
fixup 99a7b7e Add search_gem
fixup 8a98b9f Add footer & status
fixup d6cbe67 Add uptime
fixup 8cc2566 Add source_code
fixup 1b4c8ed Add data
fixup e723f99 Add discussion_forum
fixup be19758 Add statistics
fixup 1e2fc13 Add blog
fixup 7e0804a Add about
fixup d37ad3c Add help
fixup 059ea95 Add guides
fixup 4e82a4a Add contribute
fixup aecc41f Add supported_by, hosted_by, designed_by, esolved_with, optimized_by, tracking_by, monitored_by & gems_served_by
fixup 529a85d Add pages, about & title
fixup 1c5d3dd Add purpose, header, better_api, transparent_pages & enable_community
fixup cca7ebd Add founding, support & technical
fixup 173ae19 Add passwords, edit, title, submit, new, title, will_email_notice & submit
fixup 0d1bb8e Add profiles
fixup e8ee2b5 Add rubygems
fixup d321532 Add searches
fixup 763c79b Add sessions
fixup e764056 Add stats
fixup 3e442ec Translate the rest
fixup 07709db Correct grammer for subtitle & please_sign_up
fixup 0a3ba23 Correct this_rubygem_could_not_be_found
fixup 905061a Correct no_subscriptions
fixup 1a3d362 Correct gems_served_by
fixup f00b18e Correct header
fixup 51a5807 Correct support
fixup f8151b9 Correct technical
fixup 408811d Correct hide_email
fixup cb4ae9e Correct credentials
fixup 291eb96 Correct not_hosted_notice
fixup 2972f14 Correct confirm_reset
fixup ad2b344 Correct owners_header
fixup b379d93 Correct total_users
fixup c08e09a Correct have_account
fixup b915305 Correct handle
fixup f4dffea Correct who

# Rebase 21c37d9..f4dffea onto 21c37d9 (71 command(s))
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

sferik commented on config/locales/de.yml:
```diff
@@ -0,0 +1,204 @@
+de:
```
`de` should be lower case.

SuriyaaKudoIsc commented:
@sferik: I'm finished. :+1:

sferik commented:
Dankeschön!

SuriyaaKudoIsc commented:
:+1:
